src/mame/itech/itech8_m.cpp

```diff
@@ -558,40 +569,38 @@ void slikshot_state::machine_start()
 
 /*************************************
  *
- *  SCREEN_UPDATE( slikshot )
+ *  slikshot_state::screen_update
  *
  *************************************/
 
 u32 slikshot_state::screen_update(screen_device &screen, bitmap_rgb32 &bitmap, const rectangle &cliprect)
 {
-	int totaldy, totaldx;
-	int temp, i;
-
-	/* draw the normal video first */
+	// draw the normal video first
 	screen_update_2page(screen, bitmap, cliprect);
 
-	/* add the current X,Y positions to the list */
+	// add the current X,Y positions to the list
 	m_xbuffer[m_ybuffer_next % YBUFFER_COUNT] = m_fakex->read();
 	m_ybuffer[m_ybuffer_next % YBUFFER_COUNT] = m_fakey->read();
 	m_ybuffer_next++;
 
-	/* determine where to draw the starting point */
+	// determine where to draw the starting point
 	m_curxpos += m_xbuffer[(m_ybuffer_next + 1) % YBUFFER_COUNT];
 	if (m_curxpos < -0x80) m_curxpos = -0x80;
 	if (m_curxpos >  0x80) m_curxpos =  0x80;
 
-	/* compute the total X/Y movement */
-	totaldx = totaldy = 0;
-	for (i = 0; i < YBUFFER_COUNT - 1; i++)
+	// compute the total X/Y movement
+	s32 totaldx = 0, totaldy = 0;
+	for (int i = 0; i < YBUFFER_COUNT - 1; i++)
 	{
 		totaldx += m_xbuffer[(m_ybuffer_next + i + 1) % YBUFFER_COUNT];
 		totaldy += m_ybuffer[(m_ybuffer_next + i + 1) % YBUFFER_COUNT];
 	}
 
-	/* if the shoot button is pressed, fire away */
+	// if the shoot button is pressed, fire away
 	if (totaldy < m_last_ytotal && m_last_ytotal > 50 && m_crosshair_vis)
 	{
-		/* compute the updated values */
+		int temp;
+		// compute the updated values
 		temp = totaldx;
 		if (temp <= -0x80) temp = -0x7f;
 		if (temp >=  0x80) temp =  0x7f;
```

Comment:
Can you simplify this, and avoid the need for `temp` using `std::clamp`?  Something like:
```c++
		m_curvx = std::clamp<int>(totaldx, -0x7f, 0x7f);
		m_curvy = std::clamp<int>(m_last_ytotal - 50, 0x10, 0x7f);
		m_curx = std::clamp<int>(0x60 + (m_curxpos * 0x30 / 0x80), 0x30, 0x90);
```

The expression for `m_curx` looks like it could be further simplified to something like:
```c++
		m_curx = std::clamp<int>(0x60 + ((m_curxpos * 3) >> 3), 0x30, 0x90);
```